plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -888,6 +890,286 @@ public void testSystemSchemas()
+        Document document = new Document()
+                .append("_id", new ObjectId("5126bbf64aed4daf9e2ab771"))
+                .append("col1", "foo")
+                .append("parent", new Document().append("id", dbRef));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        assertThat(query("SELECT parent.id, parent.id.id FROM test." + tableName))
+                .skippingTypesCheck()
+                .matches("SELECT row('test', 'creators', %1$s), %1$s".formatted(expectedValue))
+                .isFullyPushedDown();
+        assertQuery(
+                "SELECT typeof(parent.id), typeof(parent.id.id) FROM test." + tableName,
+                "SELECT 'row(databaseName varchar, collectionName varchar, id %1$s)', '%1$s'".formatted(expectedType));
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test(dataProvider = "dbRefProvider")
+    public void testProjectionPushdownWithDBRefLikeDocument(Object objectId, String expectedValue, String expectedType)
+    {
+        String tableName = "test_projection_pushdown_with_dbref_like_document_" + randomNameSuffix();
+
+        Document document1 = new Document()
+                .append("_id", new ObjectId("5126bbf64aed4daf9e2ab772"))
+                .append("creator", new DBRef("dbref_test", "dbref_creators", objectId));
+        Document document2 = new Document()
+                .append("_id", new ObjectId("5126bbf64aed4daf9e2ab771"))
+                .append("creator", new Document().append("databaseName", "doc_test").append("collectionName", "doc_creators").append("id", objectId));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document1);
+        client.getDatabase("test").getCollection(tableName).insertOne(document2);
+
+        assertThat(query("SELECT * FROM test." + tableName))
+                .skippingTypesCheck()
+                .matches("VALUES" +
+                        " row(row(varchar 'dbref_test', varchar 'dbref_creators', " + expectedValue + "))," +
+                        " row(row(varchar 'doc_test', varchar 'doc_creators', " + expectedValue + "))")
+                .isFullyPushedDown();
+
+        assertUpdate("DROP TABLE test." + tableName);
+    }
+
+    @Test
+    public void testPredicateOnDBRefLikeDocument()
+    {
+        String tableName = "test_predicate_on_dbref_like_document_" + randomNameSuffix();
+
+        Document document = new Document()
+                .append("_id", new ObjectId("5126bbf64aed4daf9e2ab771"))
+                .append("creator", new Document().append("databaseName", "5126bc054aed4daf9e2ab772").append("collectionName", "test").append("id", "creators"));
+
+        client.getDatabase("test").getCollection(tableName).insertOne(document);
+
+        // creator of the document has same type as DBRef type, so it is getting treated as DBRef.
+        // filter will become creator.$ref and returns empty result
```

Comment:
This is unexpected behavior. I don't think we want to allow this.

Reply:
Modified `applyFilter` to not support predicate pushdown on DBRef field. Added test case as well.